pywbem_mock/_baseprovider.py

```diff
@@ -401,10 +401,10 @@ class are returned. `None` means not supplied by client and the
 
         # local_only server default is True so True or None remove properties
         if local_only is True or local_only is None:
-            for prop, pvalue in klass.properties.items():
+            for prop, pvalue in list(klass.properties.items()):
```

Comment:
Would we be more consistent here if we went to six iteritemss for consistency.

This is probably my personal bete noir but the whole move of items/values itertems/itervalues between python 2 and 3 was a real mess and being forced to add list(..) to get the correct answer just adds to their mess.

Note that it is all my personal issue but I spend a lot of time relooking up the correct implementation for exactly this.

Reply:
In this loop, we are deleting items while iterating, and six.iteritems() returns an iterator on py2 and a dictionary view on py3. I'm pretty sure that neither of them guarantees to work if items are deleted while it iterates. In fact, the dictionary view raises RuntimeError if it detects that (I saw that happen at this code here before I added the list() around it). So if we used six.iteritems(), it would still need the list() around it to stabilize the iteration. In fact, for py3, the list() is needed in any case, because both items() and six.iteritems() returns the dictionary view.

Reply:
If you want to know what is most future-safe, it is definitely using values(), items(), keys().

They return a dictionary view object in python 3 which is basically an iterator that can be used to iterate multiple times. I think it is backwards compatible to the iterators returned by python 2's iter...().

Reply:
TODO(Andy): Add comment explaining why we use list() in this case.

Reply:
DONE - added comment to all places that require using list() because of deleting while iterating.